src/components/PopupWithForm.js

```diff
@@ -6,12 +6,21 @@ export default class PopupWithForm extends Popup {
     this._popupForm = this._modalElement.querySelector(".modal__form");
     this._modalInputs = this._popupForm.querySelectorAll(".modal__input");
     this._handleFormSubmit = handleFormSubmit;
+    this.subButton = this._popupForm.querySelector(".modal__button");
+    this.subButtonText = this.subButton.textContent;
   }
 
   _handleSubmit = (evt) => {
     evt.preventDefault();
-    this._handleFormSubmit(this._getInputValues());
-    this.closeModal();
+    this.renderLoading(true);
+
+    this._handleFormSubmit(this._getInputValues())
+      .then(() => {
+        this.closeModal();
+      })
```

Comment:
Please, add `catch` here

You need to `catch` possible errors at the very end of any server request.

Reply:
Added at line 21. No new errors